log/default_logger.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  buildConsoleLogger,
+  buildFileLogger,
+  buildMultiLogger,
+} from "./builtin_loggers.ts";
+import { buildLogger, Logger } from "./logger.ts";
+
+/** Default log levels used for the default logger */
+export const defaultLogLevels = {
+  trace: 1,
+  debug: 2,
+  info: 3,
+  warn: 4,
+  error: 5,
+};
+
+/** Default log levels type used for the default logger */
+export type DefaultLogLevels = typeof defaultLogLevels;
+
+type DefaultLogger = Logger<
+    DefaultLogLevels,
+    string,
+    { source?: string } & Record<string, unknown>
+>;
+
+const defaultLoggerConsumers: DefaultLogger[] = [];
+
+const defaultLogger = buildMultiLogger(
+  defaultLogLevels,
+  defaultLoggerConsumers,
+);
+
+/**
+ * Registers a logger following the default logger types to get passed messages sent to the default logger (`log`).
+ *
+ * Example:
+ *
+ * ```typescript
+ * import { log, addDefaultLogger, buildDefaultConsoleLogger } from "https://deno.land/std/@STD_VERSION/log/mod.ts"
+ *
+ * addDefaultLogger(
+ *   buildDefaultConsoleLogger("info")
+ * )
+ *
+ * log.warn("Foo") // Will print "foo" through the registered logger above
+ * ```
+ */
+export function addDefaultLogger(
+  logger: Logger<DefaultLogLevels, string, unknown>,
+) {
+  defaultLoggerConsumers.push(logger);
+}
+
+/**
+ * The default logger. Does nothing by default, as it is ane empty multi logger - use `addDefaultLogger` to add loggers to handle it's meessages.
+ *
+ * If you are writing an application, consider using this if you want to use th default log levels.
+ *
+ * If you are a framework and want to log for your consumers, use this to allow them to handle your messages in a standardized way.
+ *
+ * Example:
+ *
+ * ```typescript
+ * import { log } from "https://deno.land/std/@STD_VERSION/log/mod.ts"
+ *
+ * log.info('Some messge')
+ * log.debug('A debug message')
+ * log.warn('Warning')
+ * ```
+ */
+export const log = defaultLogger;
```

Comment:
This is a bit surprising that it doesn't even use `console.log` under the hood; could you elaborate on the decision @LionC?

Reply:
That is one of the bigger design decisions. Currently, the global default logger does not do anything - it is essentially a multi logger that references an (unexported) top level array which loggers can be pushed into via very strict functions.

That means that if you want to use the global default logger, for it to do stdout logging, you need to do

```typescript
import {
  addDefaultLogger,
  buildDefaultConsoleLogger,
} from "std/log/mod.ts";

addDefaultLogger(
  buildDefaultConsoleLogger("info")
)
```

to set it up (that could obviously be wrapped in an even shorter function, but that is not the important decision).

The alternative would be for that to be the default, so for the default logger to start out with a console logger in its array. But that would mean that we need a way for users to remove that if they don't want it and a way to change its threshold. That is both possible, we just need to do it and it is a different experience. It widens the amount of central state and the amount of ways that state can be modified from an "append only logger list" to "logger list that can be added to and removed from and some settings for a special value in that list". That felt more complex to me, but I also see the beauty in having a default logger that just...logs without any setup at all.

To adress the obvious alternative: I would not want to remove the ability to add loggers to the default logger set, as that enables users to set up a log file / some log reporting code which will receive all messages (especially from frameworks), which I think it is very important.

TL;DR: I have no strong feelings either way